test/c_api/database_test.cpp

```diff
@@ -100,3 +102,57 @@ TEST_F(CApiDatabaseTest, CreationHomeDir) {
     std::filesystem::remove_all(homePath + "/ku_test.db");
 }
 #endif
+
+TEST_F(CApiDatabaseTest, VirtualFileSystemDeleteFiles) {
+    std::string homeDir = "/tmp/dbHome/";
+    kuzu::common::VirtualFileSystem vfs(homeDir);
+    std::filesystem::create_directories("/tmp/test1");
+    std::filesystem::create_directories("/tmp/dbHome/test1");
+
+    // Attempt to delete files outside the home directory (should error)
+    try {
+        vfs.removeFileIfExists("/tmp/test1");
+    } catch (const kuzu::common::IOException& e) {
+        // Expected behavior
+        EXPECT_STREQ(e.what(), "IO exception: Error: Path /tmp/test1 is not within the allowed "
+                               "home directory /tmp/dbHome/");
+    }
+
+    vfs.removeFileIfExists("/tmp/dbHome/test1");
+
+    ASSERT_TRUE(std::filesystem::exists("/tmp/test1"));
+    ASSERT_FALSE(std::filesystem::exists("/tmp/dbHome/test1"));
+
+    // Cleanup: Remove directories after the test
+    std::filesystem::remove_all("/tmp/test1");
+    std::filesystem::remove_all("/tmp/dbHome/test1");
+}
+
+TEST_F(CApiDatabaseTest, VirtualFileSystemDeleteFilesEdge) {
+    std::string homeDir = "/tmp/dbHome/";
+    kuzu::common::VirtualFileSystem vfs(homeDir);
+    std::filesystem::create_directories("/tmp/dbHome/../test2");
+    std::filesystem::create_directories("/tmp/dbHome/test2");
+
+    std::cout << "Resolved Path: " << std::filesystem::absolute("/tmp/dbHome/../test2")
+              << std::endl;
+    std::cout << "Exists: " << std::filesystem::exists("/tmp/test2") << std::endl;
+
+    // Attempt to delete files outside the home directory (should error)
+    try {
+        vfs.removeFileIfExists("/tmp/dbHome/../test2");
+    } catch (const kuzu::common::IOException& e) {
+        // Expected behavior
+        EXPECT_STREQ(e.what(), "IO exception: Error: Path /tmp/dbHome/../test2 is not within the "
+                               "allowed home directory /tmp/dbHome/");
+    }
+
+    vfs.removeFileIfExists("/tmp/dbHome/test2");
+
+    ASSERT_TRUE(std::filesystem::exists("/tmp/test2"));
+    ASSERT_FALSE(std::filesystem::exists("/tmp/dbHome/test2"));
+
+    // Cleanup: Remove directories after the test
+    std::filesystem::remove_all("/tmp/test2");
+    std::filesystem::remove_all("/tmp/dbHome/test2");
```

Comment:
Can you also add those test cases:
1. Delete directories with edge cases.
2. Test whether it works as expected on windows since windows paths can have a mixture of '\\' and '/'.
For example: homedir:  "C:\Desktop\dir"
Path to delete: "C:\Desktop/dir/test1" 
3. Test whether we error in wild card pattern paths. (e.g. path to delete: '/tmp/kuzu*.test')
4. Test when we delete a path with homedirectory symbol (e.g. path to delete "~/test")

Reply:
add more tests, we do not support wild card patter for safety.